src/constructs/loadbalancing/alb.test.ts

```diff
@@ -83,36 +72,21 @@ describe("The GuApplicationTargetGroup class", () => {
     publicSubnetIds: [""],
   });
 
-  test("overrides the id if the prop is true", () => {
-    const stack = simpleGuStackForTesting();
-    new GuApplicationTargetGroup(stack, "ApplicationTargetGroup", { vpc, overrideId: true });
-
-    const json = SynthUtils.toCloudFormation(stack) as SynthedStack;
-    expect(Object.keys(json.Resources)).toContain("ApplicationTargetGroup");
-  });
-
-  test("does not override the id if the prop is false", () => {
-    const stack = simpleGuStackForTesting();
-    new GuApplicationTargetGroup(stack, "ApplicationTargetGroup", { vpc });
-
-    const json = SynthUtils.toCloudFormation(stack) as SynthedStack;
-    expect(Object.keys(json.Resources)).not.toContain("ApplicationTargetGroup");
-  });
-
-  test("overrides the id if the stack migrated value is true", () => {
+  test("overrides the logicalId when existingLogicalId is set in a migrating stack", () => {
     const stack = simpleGuStackForTesting({ migratedFromCloudFormation: true });
-    new GuApplicationTargetGroup(stack, "ApplicationTargetGroup", { vpc });
+    new GuApplicationTargetGroup(stack, "ApplicationTargetGroup", { vpc, existingLogicalId: "ApplicationTargetGrp" });
 
-    const json = SynthUtils.toCloudFormation(stack) as SynthedStack;
-    expect(Object.keys(json.Resources)).toContain("ApplicationTargetGroup");
+    expect(stack).toHaveResourceOfTypeAndLogicalId("AWS::ElasticLoadBalancingV2::TargetGroup", "ApplicationTargetGrp");
   });
 
-  test("does not override the id if the stack migrated value is true but the override id value is false", () => {
+  test("does not override the id if the stack migrated value is true but existingLogicalId is not set", () => {
```

Comment:
I think that the behaviour under test is now the responsibility of [this code](https://github.com/guardian/cdk/blob/49e0cb06c9ee1f1ae86a0f50522e0ce7db757387/src/constructs/core/migrating.ts#L30-L68). I think it is covered sufficiently by [this test](https://github.com/guardian/cdk/blob/49e0cb06c9ee1f1ae86a0f50522e0ce7db757387/src/constructs/core/migrating.test.ts#L64-L78), so I would suggest removing tests like this from other constructs.

Reply:
I thought about that too 😄 . I opted to keep the tests here as a way of ensuring the magical `GuMigratingResource.setLogicalId(this, scope, props);` line is called in a construct's constructor as I forgot a few of these 😅 .

Maybe the test should be that `GuMigratingResource.setLogicalId` is called instead?

Reply:
Also just realised `GuMigratingResource.setLogicalId(this, scope, props);` doesn't take the lessons from my previous mistake onboard! Will adjust that.

Reply:
I see what you mean now, we could easily forget to call the function/remove the line accidentally during a refactor and that would have serious consequences. Perhaps we should leave these tests in for the purposes of this PR?

I don't think we need to tackle this in the current PR, but I wonder if there is a more general way to catch this type of problem across constructs... In theory any class which extends `GuMigratingResource` should call `GuMigratingResource.setLogicalId(this, scope, props)` as part of its constructor, right? I wonder if we could catch failures to do so with an assertion or lint rule.